smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/rules/ContentPublishRule.java

```diff
@@ -0,0 +1,299 @@
+    }
+
+    /**
+     * Initialize the replication client
+     * Assert publish agent is available
+     * Assert publish agent queues not blocked
+     * 
+     * @throws Exception if exception occurs
+     */
+    @Override
+    protected void before() throws Exception {
+        replicationClient = getAuthorClient().adaptTo(ReplicationClient.class);
+        doReplicationChecks();
+    }
+
+    /**
+     * The client to use for page operations. The default implementation creates a {@link CQClient}.
+     * The default implementation also uses the default admin user.
+     *
+     * @return The client to use for page operations.
+     */
+    protected CQClient getPublishClient() {
+        if (publishClient == null) {
+            publishClient = publishRule.getAdminClient(CQClient.class);
+        }
+        return publishClient;
+    }
+
+    /**
+     * The author client to use for page operations. The default implementation creates a {@link CQClient}.
+     * The default implementation also uses the default admin user.
+     *
+     * @return The client to use for page operations.
+     */
+    protected CQClient getAuthorClient() {
+        if (authorClient == null) {
+            authorClient =  authorRule.getAdminClient(CQClient.class);
+        }
+        return authorClient;
+    }
+
+    private void checkPage(final int expectedStatus) throws Exception {
+        checkPage(true, expectedStatus);
+    }
+    
+    /**
+     * Checks that a GET on the page on publish has the {{expectedStatus}} in the response
+     *
+     * @throws Exception if an error occurred
+     * @return
+     */
+    private void checkPage(boolean skipDispatcherCache, final int expectedStatus) throws PublishException {
+        final String path = root.getPath() + ".html";
+        log.info("Checking page {} returns status {}", getPublishClient().getUrl(path), expectedStatus);
+        String errorMessage = String.format("Failed to check the page %s via the AEM publish ingress (expected status %s). "
+            + "Please ensure that the CDN and Dispatcher configurations allow fetching the page.", path, expectedStatus);
+
+        try {
+            SlingHttpResponse res = null;
+            final List<NameValuePair> queryParams = skipDispatcherCache
+                ? Collections.singletonList(
+                new BasicNameValuePair("timestamp", String.valueOf(System.currentTimeMillis())))
+                : Collections.emptyList();
+            
+            URI uri = getPublishClient().getUrl(path, queryParams);
+            HttpUriRequest request = new HttpGet(uri);
+            // Disable following redirects
+            request.setParams(new BasicHttpParams().setParameter(HANDLE_REDIRECTS, false));
+            
+            res  = getPublishClient().doStreamRequest(request, null);
+            
+            // Special handling for 401,403, logging for 301,302
+            if (null != res && (res.getStatusLine().getStatusCode() == SC_UNAUTHORIZED
```

Comment:
what happens with the test execution if res is `null`?
You could check for null and fail and you wouldn't have to check for null in all the `if`s